File: app/src/main/java/rusha/x/recipe/details/RecipeDetailsViewModel.java
package rusha.x.recipe.details;

import androidx.lifecycle.MutableLiveData;

import java.util.List;

import javax.inject.Inject;

import rusha.x.BaseViewModel;
import rusha.x.SingleLiveEvent;
import rusha.x.product.Product;
import rusha.x.recipe.Recipe;
import rusha.x.recipe.RecipeIngredient;

public class RecipeDetailsViewModel extends BaseViewModel {
    @Inject
    public RecipeDetailsViewModel() {
        super();
    }

    public MutableLiveData<String> recipeNameLiveData = new MutableLiveData<>("");
    public MutableLiveData<List<RecipeIngredient>> ingredientsLiveData = new MutableLiveData<>();
    public SingleLiveEvent<Product> goToProductDetails = new SingleLiveEvent<>();

    public void init(Recipe recipe) {
        recipeNameLiveData.setValue(recipe.name);
        ingredientsLiveData.setValue(recipe.ingredients);
    }

    public void onIngredientClick(RecipeIngredient ingredient) {
        goToProductDetails.setValue(ingredient.product);
    }
}

File: app/src/main/java/rusha/x/recipe/details/RecipeDetailsViewModel.kt
package rusha.x.recipe.details

import androidx.lifecycle.MutableLiveData
import rusha.x.BaseViewModel
import rusha.x.SingleLiveEvent
import rusha.x.product.Product
import rusha.x.recipe.Recipe

class RecipeDetailsViewModel : BaseViewModel() {
    val recipeNameLiveData = MutableLiveData<String>("")
    val ingredientsLiveData = MutableLiveData<List<Recipe.Ingredient>>()
    val goToProductDetails = SingleLiveEvent<Product>()

    fun init(recipe: Recipe) {
        recipeNameLiveData.value = recipe.name
        ingredientsLiveData.value = recipe.ingredients
    }

    fun onIngredientClick(ingredient: Recipe.Ingredient) {
        goToProductDetails.value = ingredient.product
    }
}